Adding new isBotInstalledInRoom function on bdk

diogo-pessoa opened the pull request:
New function: isBotInstalledInRoom
 Included on the bdk-server, didn't feel like the best place, but I didn't see any other.

Commit:
Adding new isBotInstalledInRoom function on bdk

LongweiDeng commented on refocus-bdk-server.js:
```diff
@@ -485,6 +485,22 @@ module.exports = (config) => {
+    const botsInRoom = event.Room.RoomType.Bots;
+    const botsInstalled = [];
+    botsInRoom.forEach((bots) => botsInstalled.push(bots.name));
+    log.info(`bots installed in room: ${botsInstalled}`);
```
I'd suggest avoiding `log.info` at bdk level unless it's absolutely necessary. 
For debugging purpose, we can perhaps use `log.debug`. 
Let me know what others think.

cassiodias replied:
yep, debbug will destroy vm space :)

diogo-pessoa replied:
Thank you  @LongweiDeng, that was one of the things that I wanted to pass by you guys, the verbosity of Logs. I don't have strong feeling that this log needed, but I did use it for debugging purposes, is it okay, if I leave it as log.debug? Let me know otherwise.

cassiodias commented on refocus-bdk-server.js:
```diff
@@ -485,6 +485,22 @@ module.exports = (config) => {
+   * @param {String} bot - botName
+   * @returns {boolean}
+   */
+  function isBotInstalledInRoom(event, bot) {
```
Since you are already looping the botsInRoom, you could by the end of the loop - in the worst case - know if its there or not, then you don't need the include.

I'd suggest find something like that (don't know if it works, didn't run that :P):

```
BOT_NAME = 'botA';
botsInRoom = [{ name:"botA" },{ name:"botA" }];

let obj = botsInRoom.find(o => o.name === BOT_NAME);
if (obj) 
  return true
else
  return false
```

cassiodias replied:
or `return botsInRoom.find(o => o.name === BOT_NAME) ? true : false;`

diogo-pessoa replied:
cool, I was looking for a way of not needing the extra variable `botsInstalled = []`, maybe that will do, I'll experiment with that.

cassiodias commented on tests/isBotInstalled.js:
```diff
@@ -0,0 +1,44 @@
+};
+const bot = 'test-bot';
+
+describe('check if bot is installed in Room:', () => {
```
can you add another case when it's not in the list?

cassiodias commented on tests/isBotInstalled.js:
```diff
@@ -0,0 +1,44 @@
+
+describe('check if bot is installed in Room:', () => {
+  it('Ok, returns true if bot in RoomType is listed', () => {
+    const action = bdk.isBotInstalledInRoom(event, bot), expectation = true;
```
I'd suggest for test clarity to change action to something like isBotInstalled and remove expectations and adding true to the equal

expect(isBotInstalled).to.be.equal(true) // sounds clear reading just this line...

Commit:
Pushing changes suggested on PR

- Added test case for when bot is not on event
- remove log.info
- using .find() when looping for bot
- adding the return false explicitly, otherwise funtion returns `undefined`

cassiodias reviewed (approved):
well done!

BrandonMurnane reviewed (approved):
looks good with the changes

Commit:
updating version on Package.json
